Add `scope service` command

`scope service cribl -c tls://in.instance.whatever:10090` should:
* create the `/etc/scope/cribl/` directory 
* create `/etc/scope/cribl/scope.yml` with sane defaults and the backend settings
* create `/etc/scope/cribl/scope_protocol.yml` with defaults
* extract `libscope.so` to _(where?)_
* edit the `cribl boot-start`-installed `/etc/systemd/system/cribl.service` file to add:  
  * `Environment=LD_PRELOAD=/.../libscope.so`
  * `Environment=SCOPE_HOME=/etc/scope/cribl/`

Tried to scope a service and got a segfault. Re-opening this issue.

sudo ~/scope service zentyal -c tcp://10.0.109.237:10090

This command will make the following changes if not found already:
  - install libscope.so into /usr/lib/x86_64-linux-gnu/cribl/
  - create /usr/lib/systemd/system/zentyal.service.d/env.conf override
  - create /etc/scope/zentyal/scope.yml
  - create /var/log/scope/
  - create /var/run/scope/
Ready to proceed? [y/n]: y
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x228 pc=0x15714e4]

goroutine 1 [running]:
github.com/criblio/scope/cmd.installScope(0x7fff4ac4a8b3, 0x7, 0xc0003a621a, 0x6, 0xc0003a6275, 0x5, 0x183c0bc, 0x3, 0x0, 0x0)
	/home/ubuntu/appscope/cli/cmd/service.go:155 +0x764
github.com/criblio/scope/cmd.installSystemd(0x7fff4ac4a8b3, 0x7, 0xc0003a621a, 0x6, 0xc0003a6275, 0x5, 0x183c0bc, 0x3)
	/home/ubuntu/appscope/cli/cmd/service.go:190 +0x149
github.com/criblio/scope/cmd.glob..func15(0x2c32ae0, 0xc00037f650, 0x1, 0x3)
	/home/ubuntu/appscope/cli/cmd/service.go:56 +0x468
github.com/spf13/cobra.(*Command).execute(0x2c32ae0, 0xc00037f620, 0x3, 0x3, 0x2c32ae0, 0xc00037f620)
	/home/ubuntu/appscope/cli/.gomod/github.com/spf13/cobra@v1.1.1/command.go:854 +0x2c2
github.com/spf13/cobra.(*Command).ExecuteC(0x2c31dc0, 0x58, 0x0, 0x64)
	/home/ubuntu/appscope/cli/.gomod/github.com/spf13/cobra@v1.1.1/command.go:958 +0x375
github.com/spf13/cobra.(*Command).Execute(...)
	/home/ubuntu/appscope/cli/.gomod/github.com/spf13/cobra@v1.1.1/command.go:895
github.com/criblio/scope/cmd.Execute()
	/home/ubuntu/appscope/cli/cmd/root.go:23 +0x50
main.main()
	/home/ubuntu/appscope/cli/main.go:19 +0x93

--- a/cli/cmd/service.go
+++ b/cli/cmd/service.go
@@ -7,8 +7,9 @@ import (
 	"os"
 	"strings"
 	"syscall"
-	"github.com/criblio/scope/util"
+
 	"github.com/criblio/scope/run"
+	"github.com/criblio/scope/util"
 	"github.com/spf13/cobra"
 )
 
@@ -17,11 +18,11 @@ var serviceUser string
 
 // serviceCmd represents the service command
 var serviceCmd = &cobra.Command{
-	Use:   "service SERVICE [flags]",
-	Short: "Configure a systemd service to be scoped",
-	Long: `the "scope service" command adjusts the configuration for the named systemd service so it is scoped when it starts.`,
-Example: `scope service  cribl -c tls://in.my-instance.cribl.cloud:10090`,
-	Args: cobra.ExactArgs(1),
+	Use:     "service SERVICE [flags]",
+	Short:   "Configure a systemd service to be scoped",
+	Long:    `the "scope service" command adjusts the configuration for the named systemd service so it is scoped when it starts.`,
+	Example: `scope service  cribl -c tls://in.my-instance.cribl.cloud:10090`,
+	Args:    cobra.ExactArgs(1),
 	Run: func(cmd *cobra.Command, args []string) {
 		// must be root
 		if 0 != os.Getuid() {
@@ -35,15 +36,19 @@ Example: `scope service  cribl -c tls://in.my-instance.cribl.cloud:10090`,
 		utsname := syscall.Utsname{}
 		err := syscall.Uname(&utsname)
 		util.CheckErrSprintf(err, "error: syscall.Uname failed; %v", err)
-		buf := make ([]byte,0,64)
+		buf := make([]byte, 0, 64)
 		for _, v := range utsname.Machine[:] {
-			if v == 0 { break }
+			if v == 0 {
+				break
+			}
 			buf = append(buf, uint8(v))
 		}
 		unameMachine := string(buf)
-		buf = make ([]byte,0,64)
+		buf = make([]byte, 0, 64)
 		for _, v := range utsname.Sysname[:] {
-			if v == 0 { break }
+			if v == 0 {
+				break
+			}
 			buf = append(buf, uint8(v))
 		}
 		unameSysname := strings.ToLower(string(buf))
@@ -152,8 +157,10 @@ func installScope(serviceName string, unameMachine string, unameSysname string, 
 			err = os.Rename(configPath, examplePath)
 			util.CheckErrSprintf(err, "error: failed to move scope.yml to scope_example.yml; %v", err)
 			rc.WorkDir = configDir
-			rc.GetScopeConfig().Libscope.Log.Transport.Path = logDir + "/cribl.log"
-			rc.GetScopeConfig().Libscope.CommandDir = runDir
+			rc.SetDefault()
+			sc := rc.GetScopeConfig()
+			sc.Libscope.Log.Transport.Path = logDir + "/cribl.log"
+			sc.Libscope.CommandDir = runDir
 			err = rc.WriteScopeConfig(configPath, 0644)
 			util.CheckErrSprintf(err, "error: failed to create scope.yml: %v", err)
 		}
@@ -187,7 +194,7 @@ func installSystemd(serviceName string, unameMachine string, unameSysname string
 		}
 	}
 
-	libraryPath := installScope(serviceName, unameMachine, unameSysname, libcName);
+	libraryPath := installScope(serviceName, unameMachine, unameSysname, libcName)
 
 	overrideDir := fmt.Sprintf("%s.d", serviceFile)
 	if _, err := os.Stat(overrideDir); err != nil {
@@ -197,7 +204,7 @@ func installSystemd(serviceName string, unameMachine string, unameSysname string
 	overridePath := fmt.Sprintf("%s.d/env.conf", serviceFile)
 	if _, err := os.Stat(overridePath); err != nil {
 		content := fmt.Sprintf("[Service]\nEnvironment=LD_PRELOAD=%s\nEnvironment=SCOPE_HOME=/etc/scope/%s\n", libraryPath, serviceName)
-		err := os.WriteFile(overridePath, []byte(content), 0644)
+		err := ioutil.WriteFile(overridePath, []byte(content), 0644)
 		util.CheckErrSprintf(err, "error: failed to create override file; %v", err)
 	}
 
@@ -226,12 +233,12 @@ func installInitd(serviceName string, unameMachine string, unameSysname string, 
 		}
 	}
 
-	libraryPath := installScope(serviceName, unameMachine, unameSysname, libcName);
+	libraryPath := installScope(serviceName, unameMachine, unameSysname, libcName)
 
 	sysconfigFile := "/etc/sysconfig/" + serviceName
 	if _, err := os.Stat(sysconfigFile); err != nil {
 		content := fmt.Sprintf("LD_PRELOAD=%s\nSCOPE_HOME=/etc/scope/%s\n", libraryPath, serviceName)
-		err := os.WriteFile(sysconfigFile, []byte(content), 0644)
+		err := ioutil.WriteFile(sysconfigFile, []byte(content), 0644)
 		util.CheckErrSprintf(err, "error: failed to create sysconfig file; %v", err)
 	}
 

--- a/cli/run/scopeconfig.go
+++ b/cli/run/scopeconfig.go
@@ -96,8 +96,8 @@ type TlsConfig struct {
 	CaCertPath     string `mapstructure:"cacertpath" json:"cacertpath" yaml:"cacertpath"`
 }
 
-// setDefault sets the default scope configuration as a struct
-func (c *Config) setDefault() error {
+// SetDefault sets the default scope configuration as a struct
+func (c *Config) SetDefault() error {
 	if c.WorkDir == "" {
 		return fmt.Errorf("workDir not set")
 	}
@@ -190,7 +190,7 @@ func (c *Config) setDefault() error {
 
 // configFromRunOpts writes the scope configuration to the workdir
 func (c *Config) configFromRunOpts() error {
-	err := c.setDefault()
+	err := c.SetDefault()
 	if err != nil {
 		return err
 	}

--- a/cli/run/scopeconfig_test.go
+++ b/cli/run/scopeconfig_test.go
@@ -44,11 +44,11 @@ import (
 func TestSetDefault(t *testing.T) {
 	opt := Config{}
 	// No workdir set, should error
-	err := opt.setDefault()
+	err := opt.SetDefault()
 	assert.Error(t, err)
 
 	opt.WorkDir = "/foo"
-	err = opt.setDefault()
+	err = opt.SetDefault()
 	sc := opt.sc
 
 	testYaml := testDefaultScopeConfigYaml("/foo", 4)

